SeafarersAPI/Data/Repositories/SubmissionRepository.cs

```diff
@@ -3,56 +3,57 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
 
 namespace SeafarersAPI.Data.Repositories
 {
-    public class SubmissionRepository
+    public class SubmissionRepository : IRepository<Submission>
     {
         public SubmissionContext context;
         public SubmissionRepository()
         {
-            if(context== null)
+            if(context == null)
             {
                 context = new SubmissionContext();
             }
         }
 
-        public List<Submission> GetSubmissions()
+        public IEnumerable<Submission> GetList(int startIndex, int amountToGet)
         {
-            return context.Submissions.ToList();
+            return context.Submissions.Skip(startIndex).Take(amountToGet).ToList();
         }
 
-        public Submission GetSubmissionById(int submissionId)
+        public int Save(Submission submission)
         {
-            var submission = context.Submissions.Where(x => x.SubmissionId == submissionId).SingleOrDefault();
-
-            return submission;
+            // TODO: Add locking mechanism or we'll run into concurrency issues
+            context.Submissions.Add(submission);
+            return context.SaveChanges();
         }
 
-        public List<Submission> GetSubmissionsByCDNnumber(string cdn_number)
+        public int Delete(Submission submission)
         {
-            return context.Submissions.Where(x => x.CdnNumber == cdn_number).ToList();
+            // TODO: Add locking mechanism or we'll run into concurrency issues
+            context.Entry(submission).State = EntityState.Deleted;
+            return context.SaveChanges();
         }
 
-        public Submission GetSubmissionByConfirmationNumber(int confirmationNumber)
+        public Submission GetSubmissionById(int submissionId)
         {
-            return context.Submissions.Where(x => x.ConfirmationNumber == confirmationNumber).SingleOrDefault();
+            return this.GetSubmissionsByFilter(x => x.SubmissionId == submissionId).SingleOrDefault();
         }
 
-        public void SaveSubmission(Submission submission)
+        public IEnumerable<Submission> GetSubmissionsByCDNnumber(string cdnNumber)
         {
-            context.Submissions.Add(submission);
-            context.SaveChanges();
+            return this.GetSubmissionsByFilter(x => x.CdnNumber == cdnNumber);
         }
 
-        public void DeleteSubmissionById(int submissionId)
+        public Submission GetSubmissionByConfirmationNumber(int confirmationNumber)
         {
-            var submission = new Submission { SubmissionId = submissionId };
-
-            context.Entry(submission).State = EntityState.Deleted;
-            context.SaveChanges();
+            return this.GetSubmissionsByFilter(x => x.ConfirmationNumber == confirmationNumber).SingleOrDefault();
         }
 
+        private IEnumerable<Submission> GetSubmissionsByFilter(Func<Submission, bool> filter)
```

Comment:
To be honest I think we should make this GetByFilter in the IRepository interface and use this instead of having one for every field. We can then have the controller dealing with the appropriate filter.

Reply:
Agree.